src/components/Drawer/index.js

```diff
@@ -0,0 +1,201 @@
+            triggerRef.current = document.activeElement;
+            setDrawerState(DrawerState.OPENING);
+        }
+
+        return () => {
+            if (isOpen) {
+                CounterManager.decrement();
+                if (triggerRef.current) triggerRef.current.focus();
+                if (!CounterManager.hasModalsOpen()) {
+                    enableBodyScroll(contentElement);
+                }
+                clearAllBodyScrollLocks();
+                setDrawerState(DrawerState.CLOSING);
+            }
+        };
+    }, [isOpen]);
+
+    useEffect(() => {
+        if (drawerState === DrawerState.OPENED) {
+            drawerRef.current.focus();
+            onOpened();
+        }
+    }, [drawerState, onOpened]);
+
+    const onSlideEnd = () => {
+        if (drawerState === DrawerState.OPENING) {
+            setDrawerState(DrawerState.OPENED);
+        } else if (drawerState === DrawerState.CLOSING) {
+            setDrawerState(DrawerState.CLOSED);
+        }
+    };
+
+    const closeDrawer = () => onRequestClose();
+
+    const handleBackDropClick = event => {
+        if (isOpen && drawerRef.current.contains(event.target)) {
+            return null;
+        }
+        return closeDrawer();
+    };
+
+    const handleKeyPressed = event => {
+        event.stopPropagation();
+        if (isOpen && event.keyCode === ESCAPE_KEY) {
+            closeDrawer();
+        }
+        if (event.keyCode === TAB_KEY) {
+            manageTab(drawerRef.current, event);
+        }
+        return null;
+    };
+
+    const drawerIsOpen = [DrawerState.OPENING, DrawerState.OPENED].includes(drawerState);
+    if (drawerState !== null && drawerState !== DrawerState.CLOSED) {
+        return createPortal(
+            <StyledBackDrop
+                id={id}
+                role="presentation"
+                onClick={handleBackDropClick}
+                onKeyDown={handleKeyPressed}
+            >
+                <StyledContainer
+                    role="dialog"
+                    tabIndex={-1}
+                    aria-labelledby={headerId}
+                    aria-modal
+                    aria-hidden
+                    aria-describedby={contentId}
+                    className={className}
+                    isOpen={drawerIsOpen}
+                    style={style}
+                    size={size}
+                    slideFrom={slideFrom}
+                    ref={drawerRef}
+                    onAnimationEnd={onSlideEnd}
+                >
+                    <Header content={header} />
+                    <RenderIf isTrue={!hideCloseButton}>
+                        <StyledCloseButton
+                            icon={<CloseIcon />}
+                            title="Hide"
+                            onClick={closeDrawer}
+                        />
+                    </RenderIf>
+                    <StyledContent ref={contentRef}>{children}</StyledContent>
+                    <Footer content={footer} />
```

Comment:
the footer can be not passed, for performance reasons let's avoid render unnecessary dom element (which is costly)